### PR TITLE
Migration to React 18

### PR DESCRIPTION
**Pull request by shivam-sharma7**
## Check List (Check all the applicable boxes) 

- [ ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by shivam-sharma7** [comment]
@jesperpedersen I'm working, any feedback would be apreciated!

**Comment by johnaohara** [comment]
@shivam-sharma7 the build is currently failing with:
```
 [INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! code ERESOLVE
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! ERESOLVE could not resolve
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! 
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! While resolving: connected-react-router@6.9.3
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! Found: react@18.2.0
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! node_modules/react
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR!   react@"^18.2.0" from the root project
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR!   peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from @monaco-editor/react@4.4.6
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR!   node_modules/@monaco-editor/react
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR!     @monaco-editor/react@"^4.4.6" from the root project
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR!   16 more (@patternfly/react-core, @patternfly/react-icons, ...)
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! 
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! Could not resolve dependency:
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! peer react@"^16.4.0 || ^17.0.0" from connected-react-router@6.9.3
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! node_modules/connected-react-router
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR!   connected-react-router@"^6.9.3" from the root project
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! 
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! Conflicting peer dependency: react@17.0.2
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! node_modules/react
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR!   peer react@"^16.4.0 || ^17.0.0" from connected-react-router@6.9.3
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR!   node_modules/connected-react-router
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR!     connected-react-router@"^6.9.3" from the root project
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! 
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! Fix the upstream dependency conflict, or retry
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! this command with --force or --legacy-peer-deps
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

I think as part of this upgrade, we also need to address this issue: https://github.com/Hyperfoil/Horreum/issues/419

**Comment by jesperpedersen** [comment]
Look at why CI is failing

**Comment by shivam-sharma7** [comment]
@jesperpedersen @johnaohara what the fu*k with this now-https://github.com/Hyperfoil/Horreum/actions/runs/6243308134/job/16948525639#step:6:711

**Comment by shivam-sharma7** [comment]
See the full log
```
[WARNING] [io.quarkus.resteasy.reactive.common.deployment.ResteasyReactiveCommonProcessor] Mixing RESTEasy Reactive server and RESTEasy Classic client parts might lead to unexpected results. Consider using 'quarkus-rest-client-reactive' instead.
[INFO] Node v18.17.0 is already installed.
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] Running Quinoa package manager install command: node.exe "C:\Users\shivam sharma\Desktop\My Project\open-source\Horreum\webapp\node\node\node_modules\npm\bin\npm-cli.js" install
[WARNING] [io.quarkus.arc.processor.BeanArchives] Failed to index org.hibernate.dialect.PostgreSQL95Dialect: Class does not exist in ClassLoader QuarkusClassLoader:Deployment Class Loader: PROD@85a5fa1
[INFO] [io.quarkus.arc.processor.IndexClassLookupUtils] Class for name: org.hibernate.dialect.PostgreSQL95Dialect was not found in Jandex index. Please ensure the class is part of the index.
[WARNING] [io.smallrye.openapi.runtime.scanner.spi] SROAP07903: Duplicate operationId: RunService_addRunFromData produced by Class: io.hyperfoil.tools.horreum.api.services.RunService, Method: jakarta.ws.rs.core.Response addRunFromData(java.lang.String, java.lang.String, java.lang.String, java.lang.String, io.hyperfoil.tools.horreum.api.data.Access, java.lang.String, java.lang.String, java.lang.String, com.fasterxml.jackson.databind.JsonNode) and Class: io.hyperfoil.tools.horreum.api.services.RunService, Method: jakarta.ws.rs.core.Response addRunFromData(java.lang.String, java.lang.String, java.lang.String, java.lang.String, io.hyperfoil.tools.horreum.api.data.Access, java.lang.String, java.lang.String, java.lang.String, org.jboss.resteasy.reactive.multipart.FileUpload, org.jboss.resteasy.reactive.multipart.FileUpload)
[WARNING] [io.quarkus.deployment.steps.CombinedIndexBuildStep] Failed to index [Ljava.lang.String;: Class does not exist in ClassLoader QuarkusClassLoader:Deployment Class Loader: PROD@85a5fa1
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! Invalid Version:
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager]
[INFO] [io.quarkiverse.quinoa.deployment.packagemanager.PackageManager] npm ERR! A complete log of this run can be found in: C:\Users\shivam sharma\AppData\Local\npm-cache\_logs\2023-09-20T05_02_27_278Z-debug-0.log
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Horreum 0.8-SNAPSHOT:
[INFO]
[INFO] Horreum ............................................ SUCCESS [  0.779 s]
[INFO] Horreum infra - common ............................. SUCCESS [  3.456 s]
[INFO] Horreum Dev Services - Parent ...................... SUCCESS [  0.032 s]
[INFO] Horreum Dev Services - Runtime ..................... SUCCESS [  2.254 s]
[INFO] Horreum Dev Services - Deployment .................. SUCCESS [  5.129 s]
[INFO] horreum-api ........................................ SUCCESS [ 10.116 s]
[INFO] Horreum Backend .................................... FAILURE [ 27.581 s]
[INFO] Horreum Client ..................................... SKIPPED
[INFO] Horreum Integration Tests .......................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  51.298 s
[INFO] Finished at: 2023-09-20T10:32:30+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:3.4.0:build (default) on project horreum-backend: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkiverse.quinoa.deployment.QuinoaProcessor#prepareQuinoaDirectory threw an exception: java.lang.RuntimeException: Error in Quinoa while running package manager install command: node.exe "C:\Users\shivam sharma\Desktop\My Project\open-source\Horreum\webapp\node\node\node_modules\npm\bin\npm-cli.js" install
[ERROR]         at io.quarkiverse.quinoa.deployment.packagemanager.PackageManager.install(PackageManager.java:46)
[ERROR]         at io.quarkiverse.quinoa.deployment.QuinoaProcessor.prepareQuinoaDirectory(QuinoaProcessor.java:107)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR]         at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR]         at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR]         at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR]         at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:858)
[ERROR]         at io.quarkus.builder.BuildContext.run(BuildContext.java:282)
[ERROR]         at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
[ERROR]         at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:833)
[ERROR]         at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

is this quinoa related issue? @ia3andy

**Comment by shivam-sharma7** [comment]
@johnaohara @jesperpedersen They are not supporting for react 18.2.0 in their project this is why we are getting issue of peer dep..  
see
* monaco editor-https://github.com/suren-atoyan/monaco-react/blob/3a76077c608126458837b799941256c5b0071d15/package.json#L33
* conect-react-router- https://github.com/supasate/connected-react-router/blob/5d4c197043deb89238f9b380a6a1ef4d909f14a7/package.json#L34

WDYT?

**Comment by johnaohara** [comment]
>     * monaco editor-https://github.com/suren-atoyan/monaco-react/blob/3a76077c608126458837b799941256c5b0071d15/package.json#L33
> 
The dependency graph supports all patch release from 18.0.0 `^18.0.0`, so `18.2.0` should be ok. Do you see an error with react `18.0.2`?

>     * conect-react-router- https://github.com/supasate/connected-react-router/blob/5d4c197043deb89238f9b380a6a1ef4d909f14a7/package.json#L34
> 

This is the issue i linked previously : https://github.com/Hyperfoil/Horreum/issues/419

**Comment by shivam-sharma7** [comment]
@johnaohara I replaced `conected-react-router` with `@lagunovsky/redux-react-router` any suggetion
https://github.com/Hyperfoil/Horreum/actions/runs/6247245766/job/16959452814#step:6:717

**Comment by johnaohara** [comment]
> @johnaohara I replaced `conected-react-router` with `@lagunovsky/redux-react-router` any suggetion https://github.com/Hyperfoil/Horreum/actions/runs/6247245766/job/16959452814#step:6:717

`@lagunovsky/redux-react-router@4.3.0` requires `react-router@6.16.0`, but we are using `react-router@5.3.4`

You need to upgrade `react-router@5.3.4` in `webapp/package.json`

File: webapp/src/App.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { Component } from "react"
 import "@patternfly/patternfly/patternfly.css" //have to use this import to customize scss-variables.scss
 
 import { Nav, NavItem, NavList, Page, PageHeader, PageHeaderTools } from "@patternfly/react-core"
-import { ConnectedRouter } from "connected-react-router"
+import { ReduxRouter } from '@lagunovsky/redux-react-router'
```
Comment (johnaohara) [inline webapp/src/App.tsx:5]: with this dependency defined in `package.json` you should just be able to import ` from 'redux-react-router'` here

Comment (shivam-sharma7) [reply webapp/src/App.tsx:5]: It's not allow to import like what you suggested.

Comment (shivam-sharma7) [reply webapp/src/App.tsx:5]: @johnaohara I don't know why CI fail.

**Comment by jesperpedersen** [comment]
Remember to squash the commits once you are closer

**Comment by shivam-sharma7** [comment]
#670

**Pull request closed without merging**